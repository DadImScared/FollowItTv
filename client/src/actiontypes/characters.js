
export const ADD_CHARACTERS = 'characters/ADD_CHARACTERS';
