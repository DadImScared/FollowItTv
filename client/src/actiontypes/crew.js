
export const ADD_CREW = 'crew/ADD_CREW';
