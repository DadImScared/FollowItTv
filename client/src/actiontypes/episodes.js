
export const ADD_EPISODES = 'episodes/ADD_EPISODES';
