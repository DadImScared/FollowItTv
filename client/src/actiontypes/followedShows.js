
export const FOLLOW_SHOW = 'followedShows/FOLLOW_SHOW';
export const UNFOLLOW_SHOW = 'followedShows/UNFOLLOW_SHOW';
export const ADD_DAY = 'followedShows/ADD_DAY';
