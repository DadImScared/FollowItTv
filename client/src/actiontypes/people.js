
export const ADD_PEOPLE = 'people/ADD_PEOPLE';
