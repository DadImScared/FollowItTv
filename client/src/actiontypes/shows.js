
export const ADD_SHOW = 'shows/ADD_SHOW';
export const ADD_SHOWS = 'shows/ADD_SHOWS';
export const UPDATE_SHOW = 'shows/UPDATE_SHOW';
