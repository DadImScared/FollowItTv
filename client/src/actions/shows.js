
import axios from 'axios';

import * as showActionTypes from '../actiontypes/shows';

export const addShow = (id, payload) => {
  return {
    type: showActionTypes.ADD_SHOW,
    id,
    payload
  };
};

export const addShows = (shows) => {
  return {
    type: showActionTypes.ADD_SHOWS,
    shows
  };
};

export const getShow = async (showId) => {
  return await axios.get(`http://api.tvmaze.com/shows/${showId}`);
};
