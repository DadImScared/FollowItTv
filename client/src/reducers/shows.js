
import * as showActionTypes from '../actiontypes/shows';
import { ADD_CHARACTERS } from '../actiontypes/characters';

function shows(state={}, action) {
  switch(action.type) {
    case showActionTypes.ADD_SHOW:
      return {
        ...state,
        [action.id]: {
          ...action.payload
        }
      };
    case showActionTypes.ADD_SHOWS:
      return {
        ...state,
        ...action.shows
      };
    case ADD_CHARACTERS:
      return {
        ...state,
        [action.id]: {
          ...state[action.id],
          cast: action.characterIds
        }
      };
    case showActionTypes.UPDATE_SHOW:
      return {
        ...state,
        [action.id]: {
          ...state[action.id],
          [action.key]: action.value
        }
      };
    default:
      return state;
  }
}

export default shows;
