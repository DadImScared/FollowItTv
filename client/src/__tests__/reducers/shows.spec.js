
import * as showActionTypes from '../../actiontypes/shows';
import reducer from '../../reducers/shows';

describe('shows', () => {
  it('should add show', () => {
    const payload = { name: 'show1' };
    expect(
      reducer({},{
        type: showActionTypes.ADD_SHOW,
        id: 2,
        payload
      })
    ).toEqual({
      2: {
        ...payload
      }
    });
  });
});
