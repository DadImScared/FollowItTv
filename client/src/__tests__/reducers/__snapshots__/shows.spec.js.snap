// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows should add cast on ADD_CHARACTERS action 1`] = `
Object {
  "1": Object {
    "id": 1,
    "name": "breaking bad",
  },
  "10": Object {
    "id": 10,
    "name": "tom and jerry",
  },
  "5": Object {
    "cast": Array [
      1,
      2,
    ],
    "id": 5,
    "name": "x files",
  },
}
`;

exports[`shows should update show 1`] = `
Object {
  "1": Object {
    "id": 1,
    "name": "breaking bad",
  },
  "10": Object {
    "id": 10,
    "name": "tom and jerry",
  },
  "5": Object {
    "crew": Array [
      1,
      2,
    ],
    "id": 5,
    "name": "x files",
  },
}
`;
